### ISSUE TITLE
Harmonize Caffeine `CacheManager` bean name

### ISSUE THREAD

**Issue by wonwoo** [spring-projects/spring-boot#6980]
`CouchbaseCacheConfiguration`, `EhCacheCacheConfiguration`, `GuavaCacheConfiguration` `RedisCacheConfiguration` etc ... 
The bean names of all `cacheManager`

But CaffeineCacheConfiguration is `caffeineCacheManager`
What reason?

``` java
    @Bean
    public CaffeineCacheManager caffeineCacheManager() {
        CaffeineCacheManager cacheManager = createCacheManager();
        List<String> cacheNames = this.cacheProperties.getCacheNames();
        if (!CollectionUtils.isEmpty(cacheNames)) {
            cacheManager.setCacheNames(cacheNames);
        }
        return this.customizers.customize(cacheManager);
    }
```

**Comment by snicoll** [comment]
No reason but we should harmonise that. Thanks for the report.

**Comment by snicoll** [comment]
Given the side effect described in #7594 we should rather fix that in 1.4
